initialization/initialization.go

```diff
@@ -342,12 +361,16 @@ func (init *Initializer) initFile(computeProviderID uint, fileIndex int, numLabe
 }
 
 func (init *Initializer) updateSessionNumLabelsWritten(numLabelsWritten uint64) {
+	init.mtx.RLock()
+
 	init.numLabelsWritten = numLabelsWritten
```

Comment:
Any reason for keeping this inside of the new lock scope?

Reply:
No, it may be removed outside of the scope.